### PR TITLE
Use Cajita mesh for MPI decomposition

### LINKED ISSUE #57: Use Cajita for MPI decomposition
Replace current MPI subdomain grid/mesh with Cajita. This will simplify the `Comm` class and enable using Cajita periodic-enabled Halo/Distributors once implemented.

First step towards #17

### PR DESCRIPTION
**Pull request by streeve**
Replace MPI decomposition in `Comm` with Cajita. This will enable using Cajita particle-grid functionality in the future and enable flexibility like non-periodic systems, etc.

Closes #57

### CODE REVIEW

**Comment by streeve** [comment]
Need to update wiki when merged

File: src/system.h [hunk 1]
```diff
@@ -103,19 +110,68 @@ class SystemCommon
...
         mass = t_mass( "System::mass", ntypes );
     }
 
     ~SystemCommon() {}
 
+    void create_domain( std::array<double, 3> low_corner,
+                        std::array<double, 3> high_corner )
+    {
+        // Create the MPI partitions.
+        Cajita::UniformDimPartitioner partitioner;
```
Comment (sslattery) [inline src/system.h:132]: Are most MD simulations usually in a domain shaped like a uniform cube? If so this partitioner should always be OK.

File: src/system.h [hunk 2]
```diff
@@ -103,19 +110,68 @@ class SystemCommon
...
+        ranks_per_dim = partitioner.ranksPerDimension( MPI_COMM_WORLD, {} );
+
+        // Create global mesh of MPI partitions.
+        auto global_mesh = Cajita::createUniformGlobalMesh(
+            low_corner, high_corner, ranks_per_dim );
```
Comment (sslattery) [inline src/system.h:137]: So you are making a mesh with one "cell" per rank?

Comment (sslattery) [reply src/system.h:137]: Eventually we would like the linked cell list to be Cajita-based so I could see expanding this to be a mesh where the cells are the size of the interaction cutoff distance in the force kernel

**Review by sslattery — APPROVED** [review]

File: src/system.h (hunk 1 above)
Comment (streeve) [reply src/system.h:132]: LAMMPS has both options, but I think it makes sense to stick with this and explore the non-uniform partitioning later

File: src/system.h (hunk 2 above)
Comment (streeve) [reply src/system.h:137]: Yes, one cell per rank. But that's a good point - that will be necessary for the new grid comm as well. I'll do that separately to make sure I get the edge cases right

**Pull request merged**